tests/integration/synapseclient/models/synchronous/test_permissions.py

```diff
@@ -0,0 +1,1298 @@
+            principal_id=p1.id
+        )
+
+        # THEN I expect to see the permissions of both teams
+        expected_permissions = [
+            "READ",
+            "DELETE",
+            "CHANGE_SETTINGS",
+            "UPDATE",
+            "CHANGE_PERMISSIONS",
+            "CREATE",
+            "MODERATE",
+            "DOWNLOAD",
+        ]
+        assert set(expected_permissions) == set(permissions)
+
+    def test_get_acl_for_project_with_public_and_registered_user(self) -> None:
+        # GIVEN a project created with default permissions of administrator
+        project_with_permissions_for_public_and_authenticated_users = Project(
+            name=str(uuid.uuid4()) + "test_get_acl_for_project_with_registered_user"
+        ).store()
+        self.schedule_for_cleanup(
+            project_with_permissions_for_public_and_authenticated_users.id
+        )
+
+        # AND the user that created the project
+        p1 = UserProfile().get()
+
+        # AND the permissions for PUBLIC are set to 'READ'
+        project_with_permissions_for_public_and_authenticated_users.set_permissions(
+            principal_id=PUBLIC, access_type=["READ"]
+        )
+
+        # AND the permissions for AUTHENTICATED_USERS is set to 'READ, DOWNLOAD'
+        project_with_permissions_for_public_and_authenticated_users.set_permissions(
+            principal_id=AUTHENTICATED_USERS, access_type=["READ", "DOWNLOAD"]
+        )
+
+        # AND the permissions for the user on the entity do NOT include DOWNLOAD
+        project_with_permissions_for_public_and_authenticated_users.set_permissions(
+            principal_id=p1.id,
+            access_type=[
+                "READ",
+                "DELETE",
+                "CHANGE_SETTINGS",
+                "UPDATE",
+                "CHANGE_PERMISSIONS",
+                "CREATE",
+                "MODERATE",
+            ],
+        )
+
+        # WHEN I get the permissions for a public user on the entity
+        permissions = (
+            project_with_permissions_for_public_and_authenticated_users.get_acl()
+        )
+
+        # THEN I expect to the public permissions
+        expected_permissions = ["READ"]
+        assert set(expected_permissions) == set(permissions)
+
+        # and WHEN I get the permissions for an authenticated user on the entity
+        permissions = (
+            project_with_permissions_for_public_and_authenticated_users.get_acl(
+                principal_id=p1.id
+            )
+        )
+
+        # THEN I expect to see the permissions of the user, and the authenticated user, and the public user
+        expected_permissions = [
+            "READ",
+            "DELETE",
+            "CHANGE_SETTINGS",
+            "UPDATE",
+            "CHANGE_PERMISSIONS",
+            "CREATE",
+            "MODERATE",
+            "DOWNLOAD",
+        ]
+        assert set(expected_permissions) == set(permissions)
+
+
+class TestAclOnFolder:
```

Comment:
Would like to see a test where the local share settings on a subfolder is changed for a user and/or team, without affecting the ACL of the parent folder. Apologies if this was already written and I've missed it

Reply:
Added more tests here: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1068/commits/32f0778dd1fc097d0fcabc7cd8c108a3ec9aab5f